ksql-parser/src/main/java/io/confluent/ksql/schema/ksql/DefaultSqlValueCoercer.java

```diff
@@ -77,6 +80,34 @@
     return Optional.of(result);
   }
 
+  private static Optional<?> coerceStruct(Object value, SqlStruct targetType) {
+    if (!(value instanceof Struct)) {
+      return Optional.empty();
+    }
+
+    final Struct struct = (Struct) value;
+    final Struct coerced = new Struct(
+        SchemaConverters.sqlToConnectConverter().toConnectSchema(targetType)
+    );
+
+    for (Field field : coerced.schema().fields()) {
+      Optional<io.confluent.ksql.schema.ksql.types.Field> sqlField = targetType.field(field.name());
+      if (!sqlField.isPresent()) {
+        // if there was a field in the struct that wasn't in the schema
+        // we should throw an exception
+        return Optional.empty();
+      } else if (struct.schema().field(field.name()) == null) {
+        // if we cannot find the field in the struct, we can ignore it
```

Comment:
Seems dangerous...

Reply:
Oops! That's not what i meant to do... I meant to allow `STRUCT()` as an insert into statement for `STRUCT<foo VARCHAR>` but I don't think this condition is ever hit - there's a test for that somewhere. I'll do some more testing and get back to you.

Reply:
okay wait nevermind, this is necessary and it does exactly what I thought it does ^